WordPress/Classes/Networking/Remote Objects/RemoteBlogSettings.swift

```diff
@@ -0,0 +1,157 @@
+import Foundation
+
+
+/**
+ *  @class           RemoteBlogSettings
+ *  @brief           This class encapsulates all of the *remote* settings available for a Blog entity
+ */
+public class RemoteBlogSettings : NSObject
```

Comment:
I think the types on the remote classes should closely match their contents and not just be the types of the API or the Core Data storage. For instance:

- commentsAllowed: `Bool?`
- commentsBlackListKeys: `[String]?` or even better `Set<String>?`
- defaultCategoryID: `Int?`

The core data types should also be more like these, but there are more limitations there. I understand this might seem a bit pointless now, but when the day comes and we add support for WP-API, we don't want to be storing our data in a format decided bit a previous api design. Otherwise, we end up with little gems like [this](https://github.com/wordpress-mobile/WordPress-iOS/blob/50b0184e82a5604f73eb382c380d74d45826b279/WordPress/Classes/Networking/BlogServiceRemoteREST.m#L220) or [this](https://github.com/wordpress-mobile/WordPress-iOS/blob/50b0184e82a5604f73eb382c380d74d45826b279/WordPress/Classes/Services/PostService.m#L707).